discord/guild.py

```diff
@@ -3816,3 +3816,32 @@ async def create_auto_moderation_rule(
             self.id, payload, reason=reason
         )
         return AutoModRule(state=self._state, data=data)
+
+    async def delete_auto_moderation_rule(
+        self,
+        *,
+        id: int,
+        reason: str | None = None,
+    ) -> AutoModRule:
+        """
+        Deletes an auto moderation rule.
+
+        Parameters
+        ----------
+        id: :class:`int`
+            The ID of the auto moderation rule.
+        reason: Optional[:class:`str`]
+            The reason for deleting the rule. Shows up in the audit log.
+
+        Raises
+        ------
+        HTTPException
+            Deleting the auto moderation rule failed.
+        Forbidden
+            You do not have the Manage Guild permission.
+        """
+
+        data = await self._state.http.delete_auto_moderation_rule(
+            self.id, id, reason=reason
+        )
+        return AutoModRule(state=self._state, data=data)
```

Comment:
Deleting a rule does not return anything (status code 204)
https://discord.com/developers/docs/resources/auto-moderation#delete-auto-moderation-rule
Update the return typehint as well